src/shared/molecules/ResolvedLinkEditorPopover/ResolvedLinkEditorPopover.tsx

```diff
@@ -0,0 +1,172 @@
+  const navigate = useHistory();
+  const dispatch = useDispatch();
+  const nexus = useNexusContext();
+  const { pathname } = useLocation();
+  const routeMatch = useRouteMatch<{
+    orgLabel: string;
+    projectLabel: string;
+    resourceId: string;
+  }>(`/:orgLabel/:projectLabel/resources/:resourceId`);
+  const {
+    editorPopoverResolvedData: { open, error, resolvedAs, results },
+  } = useSelector((state: RootState) => state.uiSettings);
+  const resultPattern: TResultPattern = { resolvedAs, open };
+
+  const onClickOutside = () => {
+    dispatch({
+      type: UISettingsActionTypes.UPDATE_JSON_EDITOR_POPOVER,
+      payload: {
+        open: false,
+        top: 0,
+        left: 0,
+        error: null,
+        results: [],
+        resolvedAs: undefined,
+      },
+    });
+  };
+  const onClickLink = async (resource: TDELink) => {
+    onClickOutside();
+    if (pathname === '/data-explorer/graph-flow') {
+      dispatch(AddNewNodeDataExplorerGraphFlow(resource));
+    } else if (routeMatch?.url && routeMatch.params) {
+      const data = (await nexus.Resource.get(
+        routeMatch.params.orgLabel,
+        routeMatch.params.projectLabel,
+        routeMatch.params.resourceId
+      )) as Resource;
+      const orgProject = getOrgAndProjectFromProjectId(data._project);
+      dispatch(
+        InitNewVisitDataExplorerGraphView({
+          source: {
+            _self: data._self,
+            title: getResourceLabel(data),
+            types: getNormalizedTypes(data['@type']),
+            resource: [
+              orgProject?.orgLabel ?? '',
+              orgProject?.projectLabel ?? '',
+              data['@id'],
+            ],
+          },
+          current: resource,
+        })
+      );
+      navigate.push('/data-explorer/graph-flow');
+    }
+  };
+
+  return pmatch(resultPattern)
+    .with({ open: true, resolvedAs: 'error' }, () => (
+      <PopoverContainer {...{ onClickOutside }}>
+        <div className="popover-btn">{error}</div>
+      </PopoverContainer>
+    ))
+    .with({ open: true, resolvedAs: 'resource' }, () => {
+      const result = results as TDELink;
+      return (
+        <PopoverContainer {...{ onClickOutside }}>
+          <div className="resource" key={result._self}>
+            <Tag color="red">{`${result.resource?.[0]}/${result.resource?.[1]}`}</Tag>
+            <button
+              onClick={() => onClickLink(result)}
+              className="link popover-btn"
+            >
+              <span>{result.title ?? result.resource?.[2]}</span>
+            </button>
+          </div>
+        </PopoverContainer>
+      );
+    })
+    .with({ open: true, resolvedAs: 'resources' }, () => {
+      return (
+        <PopoverContainer {...{ onClickOutside }}>
+          {(results as TDELink[]).map(item => (
+            <div className="resource" key={item._self}>
+              <Tag color="blue">{`${item.resource?.[0]}/${item.resource?.[1]}`}</Tag>
```

Comment:
If `item.resource` is undefined what will be displayed in this div with className "resource"?

If we handle the 2 cases (where item.resource exists and does not exist) separately, we might get a few benefits:
1. We won't display an empty Tag with just a "/" in it which might be awkward.
2. It would be clearer what's rendered when.
3. We could destructure the array `item.resource` to extract the org, project, and resource id, instead of using indices:

```javascript
...map(item => {
   if (item.resource) {
   const [org, project, resourceId] = item.resource;
    return (the div with all the tags and fancy stuff)
   } else {
       // maybe just render the title?
   }
} 
```